src/Microsoft.AspNet.Mvc.Core/Formatters/JsonOutputFormatter.cs

```diff
@@ -22,6 +22,8 @@ public JsonOutputFormatter()
             SupportedMediaTypes.Add(MediaTypeHeaderValue.Parse("text/json"));
 
             _serializerSettings = new JsonSerializerSettings();
+
+            _serializerSettings.ContractResolver = new JsonContractResolver();
```

Comment:
this is answering my comment above. But still don't apply to both sides. When we do camelcasing support we can reconsider, for now its not even on the list for the 6.0 milestone